uiautomationutilities/perf_tool/MeasurementsStore.cs

```diff
@@ -0,0 +1,151 @@
+        {
+            if (!AllScenarioMeasurements.ContainsKey(scenarioName))
+            {
+                AllScenarioMeasurements.Add(scenarioName, new List<PerfMeasurementsSet>());
+                foreach (var measurementConfiguration in enabledMeasurementsConfiguration)
+                {
+                    AllScenarioMeasurements[scenarioName].Add(new PerfMeasurementsSet(measurementConfiguration));
+                }
+            }
+            foreach (var measurementSet in AllScenarioMeasurements[scenarioName])
+            {
+                measurementSet.AddIterationMeasurement(perfData);
+            }
+        }
+
+        public static void GenerateAggregateMeasurements()
+        {
+            foreach (var measurementSetList in AllScenarioMeasurements.Values)
+            {
+                //Delete any measurementSet that has no actual data.
+                measurementSetList.RemoveAll(x => x._iterationMeasurements.Count == 0);
+                measurementSetList.ForEach(t => t.GenerateAggregateMeasurements());
+            }
+        }
+
+        public static void DumpResponseTimeSummaryToFile(String fileName)
+        {
+            PerfMeasurementsSet.AppendMeasurementSummaryHeadersToFile(fileName);
+            foreach (var scenarioMeasurements in AllScenarioMeasurements)
+            {
+                AllScenarioMeasurements[scenarioMeasurements.Key].ForEach(t => t.AppendMeasurementSummaryToFile(fileName, scenarioMeasurements.Key, PerformanceMetricType.ResponseTime));
+            }
+        }
+
+        public static void DumpRssEndSummaryToFile(String fileName)
+        {
+            PerfMeasurementsSet.AppendMeasurementSummaryHeadersToFile(fileName);
+            foreach (var scenarioMeasurements in AllScenarioMeasurements)
+            {
+                AllScenarioMeasurements[scenarioMeasurements.Key].ForEach(t => t.AppendMeasurementSummaryToFile(fileName, scenarioMeasurements.Key, PerformanceMetricType.RssEnd));
+            }
+        }
+        public static void DumpVssEndSummaryToFile(String fileName)
+        {
+            PerfMeasurementsSet.AppendMeasurementSummaryHeadersToFile(fileName);
+            foreach (var scenarioMeasurements in AllScenarioMeasurements)
+            {
+                AllScenarioMeasurements[scenarioMeasurements.Key].ForEach(t => t.AppendMeasurementSummaryToFile(fileName, scenarioMeasurements.Key, PerformanceMetricType.VssEnd));
+            }
+        }
+
+        public static void DumpAllMeasurementsDataToFile(String fileName)
+        {
+            foreach (var scenarioMeasurements in AllScenarioMeasurements)
+            {
+                AllScenarioMeasurements[scenarioMeasurements.Key].ForEach(t => t.AppendMeasurementsDataToFile(fileName, scenarioMeasurements.Key));
+            }
+        }
+        /*public static void InsertMeasurementsInDb(String taskId)
```

Comment:
If we're commenting this out, either say why or remove it?

Reply:
+1